### PR TITLE
refactor: vue hooks subscribe to state

### PR DESCRIPTION
**Pull request by enesozturk**
# Description

- Refactor Vue hooks to make sure they are listen changes and rerendering the app accordingly. 
- Add multiple adapter Vue examples to test

## Type of change

- [x] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-1413
For GH issues: closes #...

# Showcase (Optional)

If there is a UI change include the screenshots with before and after state.
If new feature is being introduced, include the link to demo recording.

# Checklist

- [ ] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [ ] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: examples/vue-wagmi-solana/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    projectId: 'YOUR_PROJECT_ID',
+    debug: true // Optional - defaults to false
+  })
+  ```
+
+- [#2984](https://github.com/reown-com/appkit/pull/2984) [`b5444bd`](https://github.com/reown-com/appkit/commit/b5444bdd454d370cf3f17267d457cd46de10a5b9) Thanks [@magiziz](https://github.com/magiziz)! - Changed logger imports to not cause the webpack bundler to throw a CommonJS module error.
+
+- [#2974](https://github.com/reown-com/appkit/pull/2974) [`0563711`](https://github.com/reown-com/appkit/commit/05637111aba940af5f46215336268294675322fd) Thanks [@magiziz](https://github.com/magiziz)! - Added create wallet flow in connect modal
+
+- Updated dependencies [[`e6a36a7`](https://github.com/reown-com/appkit/commit/e6a36a746e85ced1f89fc1d7227e25df1559f063), [`b5444bd`](https://github.com/reown-com/appkit/commit/b5444bdd454d370cf3f17267d457cd46de10a5b9), [`0563711`](https://github.com/reown-com/appkit/commit/05637111aba940af5f46215336268294675322fd)]:
```
Comment (magiziz) [inline examples/vue-wagmi-solana/CHANGELOG.md:34]: Are these changelogs new ?

File: packages/appkit/exports/react.ts [hunk 2]
```diff
@@ -44,7 +44,7 @@ export type { AppKitOptions }
 // -- Hooks ------------------------------------------------------------
 export * from '../src/library/react/index.js'
 
-export function useAppKitNetwork() {
+export function useAppKitNetwork(): UseAppKitNetworkReturn {
```
Comment (magiziz) [inline packages/appkit/exports/react.ts:47]: Do we need return type here ?

**Review by magiziz — APPROVED** [review]

File: examples/vue-ethers-solana/CHANGELOG.md [hunk 3]
```diff
@@ -0,0 +1,170 @@
+# @examples/vue-ethers5
```
Comment (tomiir) [inline examples/vue-ethers-solana/CHANGELOG.md:1]: this shouldn't be here right?

File: examples/vue-wagmi-solana/CHANGELOG.md [hunk 4]
```diff
@@ -0,0 +1,170 @@
+# @examples/vue-wagmi
```
Comment (tomiir) [inline examples/vue-wagmi-solana/CHANGELOG.md:1]: same

File: packages/appkit/exports/react.ts (hunk 2 above)
Comment (tomiir) [reply packages/appkit/exports/react.ts:47]: I'd rather be explicit whenever possible so I think it's fine

File: packages/appkit/exports/vue.ts [hunk 5]
```diff
@@ -38,19 +38,27 @@ export { AppKit }
...
+  })
 
-  return {
-    caipNetwork,
-    caipNetworkId,
-    chainId,
-    switchNetwork
-  }
+  onUnmounted(() => {
+    unsubscribe?.()
```
Comment (tomiir) [inline packages/appkit/exports/vue.ts:58]: why conditional call?

**Review by tomiir — APPROVED** [review]

File: packages/appkit/exports/react.ts (hunk 2 above)
Comment (enesozturk) [reply packages/appkit/exports/react.ts:47]: Every hook or function we create should be explicit as much as this IMO. This is also very helpful for documentation and versioning

**Pull request merged**